apps/aehttp/test/aehttp_ws_test_utils.erl

```diff
@@ -120,20 +120,27 @@ init(WaitingPid) ->
 
 onconnect(_WSReq, Regs) ->
     ct:log("Ws connected"),
-    [WaitingPid] = get_registered(waiting_connected, Regs),
-    Regs1 = delete_registered(WaitingPid, waiting_connected, Regs),
-    inform_registered(WaitingPid, websocket, connected),
-    {ok, Regs1}.
+    self() ! ping,
+    {ok, Regs}.
 
 
 ondisconnect({error, {400, <<"Bad Request">>}}, Regs) ->
     {close, normal, Regs};
 ondisconnect({remote, closed}, Regs) ->
     ct:log("Connection closed, closing"),
-    {close, "closed", Regs}.
-
-websocket_handle({pong, _}, _ConnState, Regs) ->
-    {ok, Regs};
+    {close, normal, Regs}.
+
+websocket_handle({pong, Nonce}, _ConnState, Regs) ->
+    case get_registered(waiting_nonce, Regs) of
+        [Nonce] ->
+            Regs1 = delete_registered(Nonce, waiting_nonce, Regs),
+            [WaitingPid] = get_registered(waiting_connected, Regs1),
+            inform_registered(WaitingPid, websocket, connected),
+            Regs2 = delete_registered(WaitingPid, waiting_connected, Regs1),
+            {ok, Regs2};
+        _ ->
```

Comment:
Could this open match be changed to match with `[]` or anyway stricter match? (Mostly for readability - otherwise reader needs to understand why the other match is the way it is.)

Reply:
👍